src/Services/M3U.vala

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ * SPDX-FileCopyrightText: 2025 elementary, Inc. (https://elementary.io)
+ */
+
+namespace Music.M3U {
+
+    // Standard specification here: https://en.wikipedia.org/wiki/M3U
+    public File[]? parse_playlist (File playlist) {
+        debug ("Parsing playlist: %s", playlist.get_path ());
+        File[] list = {};
+
+        try {
+            FileInputStream @is = playlist.read ();
+            DataInputStream dis = new DataInputStream (@is);
+            string line;
+
+            while ((line = dis.read_line ()) != null) {
+                debug ("%s", line);
+
+                // Skip extended
+                if (line.has_prefix ("#EXT")) {
+                    debug ("Skipping EXTM3U: " + line);
+                    continue;
+                }
+
+                File target;
+
+                if (line.ascii_down ().has_prefix ("file:///")) {
+                    target = File.new_for_uri (line);
+
+                //FIXME: Music does not handle URL
+                //} else if (line.ascii_down ().has_prefix ("http")) {
+                //  target = File.new_for_url (line);
+
+                } else {
+                    target = File.new_for_path (line);
+                }
+
+                // We do not need to test yet whether files exist
```

Comment:
```suggestion
                // The caller is responsible for testing whether files exist and
                // are valid using PlaybackManager.queue_files() instead of here
```

What about like this by being explicit the reason why the test is not necessary here?